### PR TITLE
Use TB interface, rather than T pointer when doing test stuff

### PR DESCRIPTION
**Pull request by vagababov**
this will permit passing testing.B everywhere too!

/assign @markusthoemmes @dprotaso

🧹

### CODE REVIEW

File: reconciler/testing/context.go [hunk 1]
```diff
@@ -18,10 +18,10 @@ package testing
 
 import (
 	"context"
+	"testing"
 	"time"
 
 	"go.uber.org/atomic"
-	"go.uber.org/zap/zaptest"
```
Comment (evankanderson) [inline reconciler/testing/context.go:24]: Yay!

**Review by evankanderson — COMMENTED** [review]
/lgtm
/approve

**Comment by vagababov** [comment]
(ノಠ益ಠ)ノ彡┻━┻

/test pull-knative-pkg-unit-tests

**Pull request merged**